SH4RE/SH4RE/ListingViewModel.swift

```diff
@@ -273,24 +277,30 @@ func fetchSingleListing(lid:String, completion: @escaping (Listing) -> Void){
     var listing = Listing()
     
     docRef.getDocument { (document, error) in
-        let data = document!.data()
+        guard let document = document else{
+            return
+        }
+        guard let data = document.data() else{
+            return
+        }
         //Assign listing properties here
         let id = lid
-        let uid = data!["UID"] as? String ?? ""
-        let title = data!["Title"] as? String ?? ""
-        let ownerName = data!["ownerName"] as? String ?? ""
-        let description = data!["Description"] as? String ?? ""
-        let imagepath = data!["image_path"] as? [String] ?? []
-        let price = data!["Price"] as? Float ?? 0
-        let timeAvailability = data!["Availability"] as? [Timestamp] ?? []
+        let uid = data["UID"] as? String ?? ""
+        let title = data["Title"] as? String ?? ""
+        let ownerName = data["ownerName"] as? String ?? ""
+        let description = data["Description"] as? String ?? ""
+        let postalCode = data["postal"] as? String ?? ""
+        let imagepath = data["image_path"] as? [String] ?? []
+        let price = data["Price"] as? Float ?? 0
+        let timeAvailability = data["Availability"] as? [Timestamp] ?? []
         var availability:[Date] = []
-        let address = data!["_geoloc"] as? Dictionary<String,Double> ?? ["lat": -1, "long": -1]
-        let category = data!["Category"] as? String ?? ""
-        let created = data!["timestamp"] as? Timestamp ?? Timestamp(date:Date(timeIntervalSinceReferenceDate: 0))
+        let address = data["_geoloc"] as? Dictionary<String,Double> ?? ["lat": -1, "long": -1]
+        let category = data["Category"] as? String ?? ""
+        let created = data["timestamp"] as? Timestamp ?? Timestamp(date:Date(timeIntervalSinceReferenceDate: 0))
         for timestamp in timeAvailability{
             availability.append(timestamp.dateValue())
         }
-        listing = Listing(id:id,uid:uid, title:title, description:description, imagepath:imagepath, price:price, availability: availability, category: category, address: address, ownerName:ownerName, timestamp:created.dateValue())
+        listing = Listing(id:id,uid:uid, title:title, description:description, imagepath:imagepath, price:price, availability: availability, category: category, address: address, ownerName:ownerName, timestamp:created.dateValue(), postalCode: postalCode)
```

Comment:
if you are going to add postalcode to Listing(), make sure you go through all uses of "Listing(" in the app and add postal code to the examples & there are also a few uses in ListingVewModel where "Listing(" is used but postal code isn't set

Reply:
I'm pretty sure  they're all changed which cases are you referring to ?

Reply:
<img width="909" alt="Screenshot 2023-04-04 at 11 38 36 AM" src="https://user-images.githubusercontent.com/46950469/229844741-17cb1bcc-2221-4c89-9967-1486fc34a094.png">
<img width="456" alt="Screenshot 2023-04-04 at 11 39 07 AM" src="https://user-images.githubusercontent.com/46950469/229844861-de7605f6-0518-42c6-a173-2a85f61b07fd.png"> This i think this one can be removed since it was placeholder for recent listings
<img width="717" alt="Screenshot 2023-04-04 at 11 39 39 AM" src="https://user-images.githubusercontent.com/46950469/229844977-e03e7adc-322c-4a87-85c4-f2db00f73ce6.png">
<img width="546" alt="Screenshot 2023-04-04 at 11 40 01 AM" src="https://user-images.githubusercontent.com/46950469/229845064-89ff3512-525a-43c7-894c-e9a720c42467.png">

Reply:
thank you Previews shouldn't need it but I will change for the first one